kvbc/include/blockchain_view.h

```diff
@@ -74,12 +87,13 @@ class BlockchainIterator : public boost::iterator_facade<BlockchainIterator<Bloc
     }
 
     // load data from storage and, if successful, add to cache
-    auto blockInfo = BlockInfo{id};
-    blockInfo.loadIndices();
-    if constexpr (loadDataOnAccess) {
-      blockInfo.loadData();
+    if constexpr (std::is_same_v<BlockInfoInit, BlockInfoNullInit>) {
+      auto blockInfo = BlockInfo{id};
+      loadAndCacheInfo(blockInfo);
+    } else {
+      auto blockInfo = BlockInfo{id, blockInfoInit_};
+      loadAndCacheInfo(blockInfo);
```

Comment:
I guess `loadAndCacheInfo(blockInfo)` can go outside of the "if" statement

Reply:
Yes, it can. However, the intention is that we want to either impose `BlockInfo{id}` or `BlockInfo{id, blockInfoInit_}` constructors on `BlockInfo` and constructing `blockInfo` outside of the if-statement will be problematic.